### PR TITLE
Cleaning up some things

### PR DESCRIPTION
**Pull request by dignifiedquire**
WIP do not merge yet

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -374,11 +380,14 @@ function Swarm (peerInfo) {
...
       Object.keys(this.transports),
-      (key, cb) => this.transports[key].close(cb),
-      () => {
-        // Ignoring close errors
-        callback()
-      }
+      (key, cb) => {
+        // avoid unhandled error messages
+        this.transports[key].once('error', (err) => {
+          console.log('got error', err)
```
Comment (daviddias) [inline src/index.js]: let's use `debug`

File: test/05-muxing-spdy.node.js [hunk 2]
```diff
@@ -130,7 +118,8 @@ describe('stream muxing with spdy (on TCP)', function () {
   })
 
   it('close one end, make sure the other does not blow', (done) => {
-    swarmC.close(() => {
+    swarmC.close((err) => {
+      if (err) throw err
```
Comment (daviddias) [inline test/05-muxing-spdy.node.js:122]: why not the 'traditional `expect`?

File: src/index.js [hunk 3]
```diff
@@ -374,11 +380,14 @@ function Swarm (peerInfo) {
...
     async.each(
       Object.keys(this.transports),
-      (key, cb) => this.transports[key].close(cb),
-      () => {
-        // Ignoring close errors
-        callback()
-      }
+      (key, cb) => {
+        // avoid unhandled error messages
+        this.transports[key].once('error', (err) => {
```
Comment (daviddias) [inline src/index.js]:
```
Uncaught TypeError: this.transports[key].once is not a function
```

That is not an event emitter (the listener is, not the transport lib itself)

Comment (dignifiedquire) [inline src/index.js]: yeah just realised this myself, but there are other great errors :(

File: test/05-muxing-spdy.node.js (hunk 2 above)
Comment (dignifiedquire) [inline test/05-muxing-spdy.node.js:122]: I am trying to get things thrown in my face as quick as possible :D

**Pull request merged**